src/Backend/OnboardingInsuranceAPI/Areas/Insurance/AddInsurance.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using OnboardingInsuranceAPI.Services;
+
+namespace OnboardingInsuranceAPI.Areas.Insurance;
+
+public class AddInsurance: IHandler
+{
+    private readonly DataContext _context;
+
+    public AddInsurance(DataContext context)
+    {
+        _context = context; 
+    }
+
+    public async Task AddInsuranceBy(InsuranceData item)
+    {
+        var user = await _context.Users.FirstOrDefaultAsync(u => u.Pid == item.Pid);
+
+        user.CarInsurance = item.CarInsurance;
+        _context.Update(user);
```

Comment:
There is not check if user exists nor if he doesn't already have insurance. You would effectively override his insurance

Reply:
I'll implement here checking, in case user is null I'll add loger and show error like this _log.LogWarning("User is null"), user exist its ok, we are adding insurance to existing user ...